modyn/config/schema/pipeline-schema.yaml

```diff
@@ -117,22 +117,40 @@ properties:
-              presampling_ratio:
-                type: number
-                description: |
-                  [AbstractDownsampleStrategy] percentage of points on which the metric (loss, gradient norm,..) is computed. Must be between 0 and 100
-              sample_then_batch:
-                type: boolean
-                description: |
-                  [AbstractDownsampleStrategy] If True, the samples are first sampled and then batched and supplied to the training loop. If False, the datapoints are first divided into batches and then sampled.
-              downsampling_ratio:
-                type: number
+              presampling_config:
+                type: object
                 description: |
-                  [AbstractDownsampleStrategy] Ratio post_sampling_size/pre_sampling_size. For example, with 160 datapoints and a ratio of 50 we keep 80. Must be in (0,100).
-              downsampling_period:
-                type: number
+                  [CoresetStrategy] Config for the presampling strategy. If missing, no presampling is applied.
+                properties:
+                  strategy:
+                    type: string
+                    description: |
+                      Strategy used to presample the data. Available strategies: Random, Empty (get all data)
+                  ratio:
+                    type: number
+                    description: |
+                      Percentage of points on which the metric (loss, gradient norm,..) is computed. Must be between 0 and 100
+              downsampling_config:
+                type: object
                 description: |
-                  [AbstractDownsampleStrategy] In multi-epoch training and sample_then_batch, how frequently the data is selected. The default value is 1 (select every epoch). To select once per trigger, set this parameter to 0.
+                  [CoresetStrategy] Config for the downsampling strategy. If missing, no downsampling is applied.
+                properties:
+                  strategy:
+                    type: string
+                    description: |
+                       Strategy used to downsample the data. Available strategies: Loss, Gradnorm, Empty (no downsampling)
+                  sample_then_batch:
+                    type: boolean
+                    description: |
+                      If True, the samples are first sampled and then batched and supplied to the training loop. If False, the datapoints are first divided into batches and then sampled.
+                  ratio:
+                    type: number
+                    description: |
+                      Ratio post_sampling_size/pre_sampling_size. For example, with 160 datapoints and a ratio of 50 we keep 80. Must be in (0,100).
```

Comment:
it can be (0,100], right?

Reply:
Oh yes, but just with the EmptyPresampling (otherwise doing downsampling it's just a waste of time). I think it's better to leave it like that

Reply:
Sorry, I don't get your point here. 100% presampling and just using downsampling is valid, isn't it? And same would be < 100% presamling and 100% downsampling (which is identical to EmptyPresampling). So if we allow 100% for the EmptyPresampling strategy, then it should be like that in the description? Or would you never explicitly define an EmptPresamplingStrategy?

Reply:
Ok, now it is (0,100]. Previously, it did not make sense to give the option of explicitly defining Empty, now it does (as it can be part of a scheduling with something else)